Tools/Scripts/webkitpy/api_analyzer/driver.py

```diff
@@ -0,0 +1,241 @@
+                                '/usr/lib/system/libcompiler_rt*',
+                                '/usr/lib/system/libunwind*'),
+    'usr/lib/libicucore.A.tbd': (),
+}
+
+
+class TSVReporter:
+    def __init__(self, args: argparse.Namespace):
+        self.n_issues = 0
+        self.seen: set[str] = set()
+        self.print_details = args.details
+
+    def missing_selector(self, name: str, ignore=False):
+        self.seen.add(name)
+        if not ignore:
+            if self.print_details:
+                print('selector:', name, sep='\t')
+            self.n_issues += 1
+
+    def missing_class(self, name: str, ignore=False):
+        self.seen.add(name)
+        if not ignore:
+            if self.print_details:
+                print('class:', name, sep='\t')
+            self.n_issues += 1
+
+    def missing_symbol(self, name: str, ignore=False):
+        self.seen.add(name)
+        if not ignore:
+            if self.print_details:
+                print('symbol:', name, sep='\t')
+            self.n_issues += 1
+
+    def finished(self):
+        print(f'{self.n_issues} potential use(s) of SPI.')
+        if self.n_issues and not self.print_details:
+            print('Rerun with --details to see each validation issue.')
+
+
+def main(argv=None):
+    parser = argparse.ArgumentParser(description='''\
+    Using API availability information from a directory of SDKDB records,
+    scan a Mach-O binary for use of unknown symbols or Objective-C selectors.
+    ''')
+    parser.add_argument('input_files', nargs='*', type=Path,
+                        help='binaries and SDKDBs to allow arbitrary use of')
+    parser.add_argument('-a', '--arch-name', required=True,
+                        help='which architecture to analyze binary with')
+    parser.add_argument('--primary-file', help='file to analyze', type=Path)
+    parser.add_argument('--sdkdb-dir', type=Path, required=True,
+                        help='directory of partial SDKDB records for an SDK')
+    parser.add_argument('--sdkdb-cache', type=Path, required=True,
+                        help='database file to store SDKDB availabilities')
+    parser.add_argument('--sdk-dir', type=Path, required=True,
+                        help='Xcode SDK the binary is built against')
+    parser.add_argument('--depfile', type=Path,
+                        help='write inputs used for incremental rebuilds')
+    parser.add_argument('--details', action='store_true',
+                        help='print a line for each unknown symbol')
+    if additions:
+        additions.add_arguments(parser)
+    args = parser.parse_args(argv)
+
+    # For the depfile, start with the paths of all the modules in api_analyzer
+    # since this library is part of WebKit source code.
+    inputs = []
+    for info in pkgutil.walk_packages(api_analyzer.__path__):
+        # This code asssumes api_analyzer is a file-based module.
+        assert isinstance(info.module_finder, FileFinder)
```

Comment:
Do we actually care? We only care that origin is not `None`.

Reply:
I don't care, but mypy does…I'll see if I can find a better way to express it.